crates/sui-protocol-constants/src/lib.rs

```diff
@@ -51,11 +51,18 @@ pub const MAX_FIELDS_IN_STRUCT: usize = 32;
 /// Maximum dependency depth. Enforced by the Move linker when loading dependent modules.
 pub const MAX_DEPENDENCY_DEPTH: usize = 100;
 
-/// Maximum number of events that a single Move function can emit. Enforced by the Sui adapter during execution.
-// TODO: is this per Move function, or per transaction? And if per-function, can't I get around the limit by calling
-// a function that emits 255 events in a loop?
+/// Maximum number of Move events that a single transaction can emit. Enforced by the VM during execution.
 pub const MAX_NUM_EVENT_EMIT: u64 = 256;
 
+/// Maximum number of new IDs that a single transaction can create. Enforced by the VM during execution.
+pub const MAX_NUM_NEW_IDS: u64 = 1024;
```

Comment:
Small request - Can you use a more specific name for these? Since it's a global sui protocol constants file, unqualified "IDs" could mean almost anything

Reply:
I just added "OBJECT" in front of IDs, but can you think of better names please? I'm bad at naming

Reply:
I think the new names are fine, the one little tweak I'd suggest is that you should be consistent between all three about whether you name it OBJECT or MOVE_OBJECT (but I don't have a strong opinion on which).

Reply:
done